### PR TITLE
Port drone build tasks to gh workflows + actions

### PR DESCRIPTION
**Pull request by richard-cox**
Proposed changes
======

- Remove old drone build tasks (FOSSA will not be ported at this time, discussed in slack)
- create generic 'setup' composite action which can be used by multiple workflows
- create generic build-and-upload workflow which can be used by multiple workflows
  - remove dapper + use build script directly
  - tidy up build script
- create workflow to build from specific branch (triggers build-and-upload)
- create workflow to build from specific tags (triggers build-and-upload)

Linked Issues
======
 

https://github.com/rancher/dashboard/issues/10635

Further comments
======

Examples
- building from branch (with uploaded bits) - https://github.com/rancher/ui/actions/runs/8803491543
- building from tag (with uploaded bit) - https://github.com/rancher/ui/actions/runs/8803759537

rancher/dashboard pr https://github.com/rancher/dashboard/pull/10874

### CODE REVIEW

File: .github/workflows/build-branch.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+name: Build (Branch)
+on:
+  push:
+    branches:
+      - master
```
Comment (richard-cox) [inline .github/workflows/build-branch.yaml:5]: add other branches `releases-`

File: .github/workflows/build-release.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+name: Build (Release)
+on:
+  push:
+    tags:
+      - v2.9.*
```
Comment (richard-cox) [inline .github/workflows/build-release.yaml:5]: v*

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -5,9 +5,10 @@ on:
       - master
       - 'release-*'
   pull_request:
-      branches:
-        - master
-        - 'release-*'
+    branches:
+      - master
+      - 'release-*'
+  workflow_call:
```
Comment (richard-cox) [inline .github/workflows/test.yaml:11]: add comment, why here

File: scripts/ci [hunk 4]
```diff
@@ -1,46 +0,0 @@
...
-./scripts/bootstrap
-
-if [[ -z "${CI_BUILD_TAG}" ]]; then
-  echo "Linting.."
-  ./node_modules/.bin/eslint --max-warnings 0 .
-fi
-
-if [[ "${BUILD_LATEST}" == "true" ]]; then
-  echo "Building latest..."
-  ./scripts/build-static -l -v latest2
```
Comment (richard-cox) [inline scripts/ci:37]: test once merged

File: scripts/gcs_upload_asset [hunk 5]
```diff
@@ -1,7 +1,9 @@
-#!/bin/bash -x
+# #!/bin/bash -x
```
Comment (richard-cox) [inline scripts/gcs_upload_asset:1]: delete file

File: .github/workflows/build-release.yaml [hunk 6]
```diff
@@ -2,7 +2,7 @@ name: Build (Release)
 on:
   push:
     tags:
-      - v2.9.*
+      - v*
```
Comment (richard-cox) [inline .github/workflows/build-release.yaml:5]: v[2-9].*

**Review by nwmac — APPROVED** [review]
LGTM

**Pull request merged**